### ISSUE TITLE
chances of recipes in recipe tree crafting mode ignored (1.1.3+1.20.1+fabric)

### ISSUE THREAD

**Issue by MrRedstoneToGo** [emilyploszaj/emi#471]
Look at the soul sand:
![2024-03-16_02 26 20](https://github.com/emilyploszaj/emi/assets/163460667/e4fbd1bd-e7d9-4ec5-a590-77c1f04264eb)
![2024-03-16_02 26 28](https://github.com/emilyploszaj/emi/assets/163460667/3dd68dd6-5ae8-4bf8-8229-b540badf8279)
Shouldn't it be ~563 soul sand at the second picture? I think EMI ignores chances in the recipe tree crafting mode but copies the ingredients at the bottom of the recipe tree (which are marked yellow)

PS: Ignore that you can't craft 11 gold nuggets to an ingot without rest

**Comment by emilyploszaj** [comment]
This is working as intended, while it would be better to display the approximate equality there really isn't that much space, and crafting a recipe tree with chance doesn't make a ton of sense inherently.

**Comment by MrRedstoneToGo** [comment]
It's just a bit annoying. If I wash soul sand, the number in this mode just decreases by 1 instead of 9. So I have to look in the recipe tree every time to estimate the amount of sand for haunting.

**Comment by emilyploszaj** [comment]
Oh I was looking at the wrong thing, yes that's definitely an issue.

**Comment by MrRedstoneToGo** [comment]
Thanks a lot.

**Comment by MrRedstoneToGo** [comment]
Oh, and how to get a fluid ingredient green?

**Comment by MrRedstoneToGo** [comment]
I you've fixed this: Is there any way to tell EMI in the recipe tree crafting mode that you have already pruduced a fluid?

**Comment by emilyploszaj** [comment]
No, if it's not in your inventory, which is pretty difficult for a fluid to be, it'll be counted as not present. If you complete any step above that requires the fluid, it'll disappear.

**Comment by MrRedstoneToGo** [comment]
The recipe tree has one big problem: It miscalculates with chances. Another bug, this time with catalysts:
![2024-03-19_19 50 40](https://github.com/emilyploszaj/emi/assets/163460667/7d0eb6ee-8887-4976-a17d-1843ca8d9fd8)
Regardless of the output of a recipe, EMI should make the amount of catalyst liquids/items only (!) dependent on the description in the recipe.

**Comment by MrRedstoneToGo** [comment]
Since the (original) thing with the crafting mode under the favorites side bar isn't fixed yet (sorry for the mess), I looked a bit into EMI's code and it seems that you simply forgot to implement the devisor/chance at https://github.com/emilyploszaj/emi/blob/64a81e8537809b6ffd9ceef5d1a8992cd131f9f6/xplat/src/main/java/dev/emi/emi/runtime/EmiFavorites.java#L257 and https://github.com/emilyploszaj/emi/blob/64a81e8537809b6ffd9ceef5d1a8992cd131f9f6/xplat/src/main/java/dev/emi/emi/runtime/EmiFavorites.java#L270

**Comment by Twisted-Code** [comment]
Possibly related issue! (Let me know if I should open a separate issue): 
I'm getting some weird behavior when trying to use the recipe tree to craft a fletching table while the create:crushing_wheel recipe for crushing gravel is set as my default. It shows sand when the ingredient I actually want from that recipe is flint
![image](https://github.com/user-attachments/assets/4dfa8419-a51a-4530-9b10-7f58c0d560aa)

**Comment by MrRedstoneToGo** [comment]
It's a different issue... If it is a problem at all. Maybe the sand represents the recipe and not the output (the main ingredient is always shown) and that would be an improvement. I also have another very similar small problem in mind. Start another issue for that, I'll post my issue there too.

**Comment by LTCatt** [comment]
![Image](https://github.com/user-attachments/assets/27db541f-8187-455b-8c58-d225a6ed98bf)

![Image](https://github.com/user-attachments/assets/897471ab-1bb7-4208-b876-ce73b8c2b730)

Is this the same issue?

**Comment by MrRedstoneToGo** [comment]
No, this is probably a forge-related problem with JEI, as EMI cannot read the changes from it yet.

**Comment by HollyEvergreen** [comment]
<img width="1040" height="713" alt="Image" src="https://github.com/user-attachments/assets/729a8186-37c7-4b2b-9eb5-18f69392a1a3" />

<img width="768" height="713" alt="Image" src="https://github.com/user-attachments/assets/654cea43-5e25-4914-9a87-395871ac8068" />

the amount of blackstone here should be 800 but it shows as 8 (i checked the chance of the recipe in kjs it is 1%) surely the bug could be fixed by doing 
y=x/chance
where 
    y is the recipe input
    x is number of items returned from the recipe
for example in this situation y=x/c == 800=8/0.01
im not sure abt how failure chances would be handled with things like the prec mech

**Comment by BugInYourCode01** [comment]
Nope, these are not related. In fact, while rereading the (little clumsy) issue description, there are three independent problems:
1. EMI gives you the correct expected amount of base ingredients (leaf nodes), but intermediate recipes show up in the wrong quantity in the 'crafting mode' when chances are involved, and they always display their first output as an icon (dunno if that's intended)
2. Ingredients marked as 'catalysts' (no consumption) are not handled as an exception in the recipe tree – you only need the amount required for one recipe batch.
3. EMI doesn't carry over the chances in recipes from JEI (tooltip starting with 'jei:'). This is more of an API problem than a miscalculation, so it should get its own issue.

Hope I could clarify things a little :-)
